Sources/PCPayloadDeserializer.swift

```diff
@@ -192,6 +216,45 @@ struct PCPayloadDeserializer {
             size: size
         )
     }
+    
+    fileprivate static func createPartFromPayload(_ payload: [String: Any]) throws -> PCPart {
+        var partContent: String?
+        if let content = payload["content"] as? String {
+            partContent = content
+        }
+        
+        var partURL: String?
+        if let url = payload["url"] as? String {
+            partURL = url
+        }
+        
+        guard let partType = payload["type"] as? String else {
+            throw PCPayloadDeserializerError.incompleteOrInvalidPayloadToCreteEntity(type: String(describing: PCPart.self), payload: payload)
+        }
+        
+        // Empty mutable part
+        var resultPartPayload: PCMultipartPayload = PCMultipartPayload.inlinePayload(payload: PCMultipartInlinePayload(type: "", content: ""))
+        
+        var partAttachment: [String: Any]?
+        if let attachment = payload["attachment"] as? [String: Any] {
+            partAttachment = attachment
+        }
+        
+        if partContent != nil {
+            resultPartPayload = PCMultipartPayload.inlinePayload(payload: PCMultipartInlinePayload(type: partType, content: partContent!))
+        }
+        
+        if partURL != nil {
+            resultPartPayload = PCMultipartPayload.urlPayload(payload: PCMultipartURLPayload(type: partType, url: partURL!))
+        }
+        
+        if partAttachment != nil {
+            let multipartAttachment = try createMultipartAttachmentFromPayload(partAttachment!)
+            resultPartPayload = PCMultipartPayload.attachmentPayload(payload: multipartAttachment)
+        }
+        
+        return PCPart(payload: resultPartPayload)
+    }
```

Comment:
The variable assignments followed by the nil checks seems redundant. Can we just do the thing in the if block. e.g.

```swift
        if let url = payload["url"] as? String {
            return PCPart(payload: PCMultipartPayload.urlPayload(
                payload: PCMultipartURLPayload(type: partType, url: url)
            ))
        }
```

Reply:
Yeah, the compiler will complain since we still have to return something at the end of the function that don't fall in any of the branches despite the fact that we're guaranteed that one of the branches will run. Not sure there's a nicer way to organise this bit..

Reply:
Well maybe at least

```swift
        if let url = payload["url"] as? String {
            resultPartPayload = PCMultipartPayload.urlPayload(
                payload: PCMultipartURLPayload(type: partType, url: url)
            )
        }
```

then still return at the end